db/range_del_aggregator_v2.cc

```diff
@@ -214,53 +245,47 @@ void ReverseRangeDelIterator::Invalidate() {
   inactive_iters_.clear();
 }
 
-RangeDelAggregatorV2::RangeDelAggregatorV2(const InternalKeyComparator* icmp,
-                                           SequenceNumber /* upper_bound */)
-    : icmp_(icmp), forward_iter_(icmp, &iters_), reverse_iter_(icmp, &iters_) {}
-
-void RangeDelAggregatorV2::AddTombstones(
-    std::unique_ptr<FragmentedRangeTombstoneIterator> input_iter,
-    const InternalKey* smallest, const InternalKey* largest) {
-  if (input_iter == nullptr || input_iter->empty()) {
-    return;
-  }
-  if (wrapped_range_del_agg != nullptr) {
-    wrapped_range_del_agg->AddTombstones(std::move(input_iter), smallest,
-                                         largest);
-    // TODO: this eats the status of the wrapped call; may want to propagate it
-    return;
-  }
-  iters_.emplace_back(new TruncatedRangeDelIterator(std::move(input_iter),
-                                                    icmp_, smallest, largest));
-}
-
-bool RangeDelAggregatorV2::ShouldDelete(const ParsedInternalKey& parsed,
-                                        RangeDelPositioningMode mode) {
-  if (wrapped_range_del_agg != nullptr) {
-    return wrapped_range_del_agg->ShouldDelete(parsed, mode);
+bool RangeDelAggregatorV2::StripeRep::ShouldDelete(
+    const ParsedInternalKey& parsed, RangeDelPositioningMode mode) {
+  if (!InStripe(parsed.sequence)) {
```

Comment:
Are there valid cases where we call `ShouldDelete` on the wrong stripe, or should we change this to an assert?

Reply:
Probably fine to remove this check, though I'm not entirely sure. I'll check this with some stress tests.

Reply:
Sorry, I shouldn't have tried to respond quickly at home. That check is necessary during compaction, because the binary search in the StripeRep map may find an entry that isn't the same snapshot stripe as the lookup key. This is equivalent to the lazy map initialization logic from the v1 RangeDelAggregator.